### PR TITLE
bpo-42064: Convert `sqlite3` global state to module state

### PR DESCRIPTION
**Pull request by erlend-aasland**
https://bugs.python.org/issue42064

### CODE REVIEW

**Comment by encukou** [comment]
(And any subclass will necessarily be defined in another module, so `PyType_GetModule` won't return _sqlite3 for it.)

File: Modules/_sqlite/module.h [hunk 1]
```diff
@@ -63,22 +63,20 @@ typedef struct {
...
-pysqlite_get_state(PyObject *Py_UNUSED(module))
+pysqlite_get_state(PyObject *module)
 {
-    return &pysqlite_global_state;  // Replace with PyModule_GetState
+    pysqlite_state *state = (pysqlite_state *)PyModule_GetState(module);
+    assert(state != NULL);
+    return state;
 }
 
+struct PyModuleDef _sqlite3module;
```
Comment (encukou) [inline Modules/_sqlite/module.h]: This should be `extern`.

**Review by encukou — COMMENTED** [review]
Other than this, the change looks good:

File: Modules/_sqlite/module.h (hunk 1 above)
Comment (erlend-aasland) [reply Modules/_sqlite/module.h]: Yes, thanks!

Comment (erlend-aasland) [reply Modules/_sqlite/module.h]: PTAL

**Pull request merged**